### DISCUSSION TITLE
How to communicate with Metamask on mobile?

### DISCUSSION THREAD

**Discussion by olehmell** [wevm/wagmi#2549] [Q&A]
Hello devs!

I'm currently working on [Grill.chat](https://grill.chat), a dapp designed for anonymous chats. Presently, I am trying to integrate a feature that would allow you to donate directly within the chat.

I've been encountering an issue with the wagmi lib. I've been utilizing the `useConnect` hook; however, the `connectAsync() | connect()` functions don't seem to be operating as expected.

Interestingly, when using RainbowKit to connect to Metamask, it works seamlessly on the first donation attempt - Metamask opens, and everything functions properly. However, if I refresh the page and attempt to utilize `useConnect` again, it appears to lose the connection with Metamask. As a result, the donation button ceases to function - Metamask doesn't open, nor does it prompt for a transaction signature.

Is anyone able to assist me in diagnosing what's going wrong? I would sincerely appreciate your advice! 🙏💡💻

**Comment by adrianotadao** [comment]
hey @olehmell I have a similar issue, how did you handle it? thx!
